rplugin/python3/deoplete/source/file.py

```diff
@@ -66,7 +66,11 @@ def __longest_path_that_exists(self, context, input_str):
     def __substitute_path(self, context, path):
         m = re.match(r'(\.{1,2})/+', path)
         if m:
-            base = context['bufname'] if self.__buffer_path else context['cwd']
+            if self.__buffer_path and context['bufname']:
+                base = context['bufname']
+            else:
+                base = os.path.join(context['cwd'], 'x')
```

Comment:
Forgot to mention in the PR message that it also adds a path component to `cwd` so that `./` and `../` should work correctly when not using the buffer path option.